### DISCUSSION TITLE
Attachconstraint doesn't work in my programme.

### DISCUSSION THREAD

**Discussion by no240805** [sofa-framework/sofa#4277] [Develop in C++] [answered]
I try to use  Attachconstraint to attach 2 lines with indices ,but it doesn't work. I can see the connect line but the indices doesn't keep together. the photo website is 

https://tianxiaobin.oss-cn-shanghai.aliyuncs.com/img/image-20231102211430660.png

 my file is list here.

``` xml
<Node name="root" gravity="0 -9.81  0" dt="0.01" bbox="-5 -5 -5 5 5 5">
	<RequiredPlugin name="BeamAdapter"/>
	<!-- Needed to use components [WireBeamInterpolation WireRestShape] -->
	<RequiredPlugin name="Sofa.Component.Constraint.Projective"/>
	<!-- Needed to use components [FixedConstraint] -->
	<RequiredPlugin name="Sofa.Component.ODESolver.Backward"/>
	<!-- Needed to use components [EulerImplicitSolver] -->
	<RequiredPlugin name="Sofa.Component.StateContainer"/>
	<!-- Needed to use components [MechanicalObject] -->
	<RequiredPlugin name="Sofa.Component.Topology.Container.Constant"/>
	<!-- Needed to use components [MeshTopology] -->
	<RequiredPlugin name="Sofa.Component.Visual"/>
	<!-- Needed to use components [VisualStyle] -->
	<RequiredPlugin name="Sofa.Component.MechanicalLoad"/>
	<!-- Needed to use components [ConstantForceField] -->
	<RequiredPlugin name="Sofa.Component.Constraint.Lagrangian.Model"/>
	<!-- Needed to use components [BilateralInteractionConstraint] -->
	<RequiredPlugin name="Sofa.Component.Mapping.NonLinear"/>
	<!-- Needed to use components [RigidRigidMapping] -->
	<RequiredPlugin name="Sofa.Component.LinearSolver.Direct"/>
	<!-- Needed to use components [BTDLinearSolver] -->

	<DefaultAnimationLoop />
	<VisualStyle displayFlags="showBehaviorModels showCollisionModels hideBoundingCollisionModels showForceFields" />
	<Node name="BeamModelAB">

		<Node name="BeamModelA">
			<EulerImplicitSolver rayleighStiffness="0.1" rayleighMass="0.1" printLog="false" />
			<BTDLinearSolver printLog="false" verbose="false" />

			<MechanicalObject template="Rigid3" name="DOFSA" position="	
			0 0 0 0 0 0.707 0.707 
			0 1 0 0 0 0.707 0.707   
			0 2 0 0 0 0.707 0.707   
			0 3 0 0 0 0.707 0.707    
			0 4 0 0 0 0.707 0.707 " />
			<MeshTopology name="lines" lines="0 1 1 2 2 3 3 4" />
			<FixedConstraint name="FixedConstraint" indices="0" />
			<BeamInterpolation name="BeamInterpolation" radius="0.1" />
			<AdaptiveBeamForceFieldAndMass name="BeamForceField" computeMass="1" massDensity="1"/>


			<ConstantForceField indices="3" totalForce="22 100 0 0 0 0 " />
			<ConstantForceField indices="2" totalForce="2 0 0 0 0 0 " />
			<ConstantForceField indices="1" totalForce="2 0 0 0 0 0 " />
		</Node>

		<Node name="BeamModelB">
			<EulerImplicitSolver rayleighStiffness="0.1" rayleighMass="0.1" printLog="false" />
			<BTDLinearSolver printLog="false" verbose="false" />

			<MechanicalObject template="Rigid3" name="DOFSB" position="	
			0 3 0 0 0 0.707 0.707 
			0 4 0 0 0 0.707 0.707   
			0 5 0 0 0 0.707 0.707   
			0 6 0 0 0 0.707 0.707    
			0 7 0 0 0 0.707 0.707 " />
			<MeshTopology name="lines" lines="0 1 1 2 2 3 3 4" />

			<BeamInterpolation name="BeamInterpolation" radius="0.1" />
			<AdaptiveBeamForceFieldAndMass name="BeamForceField" computeMass="1" massDensity="1"/>
		</Node>

	<AttachConstraint object1="@BeamModelA" object2="@BeamModelB" twoWay="true" indices1="3 4" indices2="0 1" constraintFactor="1 1" />
	</Node>

</Node>
```

**Comment by bakpaul** [comment, marked as answer]
HI @no240805, I've copy/pasted your scene and it runs perfectly on my installation. Do you have any displayed error on the terminal when launching the scene ?  
![Boing](https://github.com/sofa-framework/sofa/assets/30337881/e8ab3ab9-5ec3-4bb5-ae8a-11ef27f3ea52)

**Reply by no240805** [reply]
i use 23.06

**Reply by no240805** [reply]
thanks， tomorrow I will re -install the sofa

**Reply by no240805** [reply]
I try it in ubuntu23.10 ,but it doesn't work as yours. 
I download and run the SOFA_v23.06.00_Linux.run .

**Reply by bakpaul** [reply]
Ok, I cannot give support on non LTS version of Ubuntu... If the problem is related to your Ubuntu version then I simply cannot try it myself. 
But before jumping into this conclusion I would need more information : can you give me the output prompted when you launch the simulation ? It should look a bit like that :
``` 
[INFO]    [runSofa] Loading automatically plugin list in /home/paul/dev/build/sofa/lib/plugin_list.conf.default
[INFO]    [PluginManager] Loaded plugin: /home/paul/dev/build/sofa/lib/libSofa.Component.ODESolver.Forward.so
[INFO]    [PluginManager] Loaded plugin: /home/paul/dev/build/sofa/lib/libSofa.Component.ODESolver.Backward.so
....
```

**Reply by no240805** [reply]
thanks,I will do this

**Reply by no240805** [reply]
this is output: 
the first part is what terminal show when i run sofa ,the last two lines is what terminal show when I open my scene. 

sofa@sofa-Lenovo-Legion-R9000P2021H:~/SOFA/v23.06.00/bin$ ./runSofa -a
[INFO]    [runSofa] PluginRepository paths = /home/sofa/SOFA/v23.06.00/plugins:/home/sofa/SOFA/v23.06.00/collections:/home/sofa/SOFA/v23.06.00/lib
[INFO]    [runSofa] DataRepository paths = /home/sofa/SOFA/v23.06.00/share/sofa:/home/sofa/SOFA/v23.06.00/share/sofa/examples:/home/sofa/SOFA/v23.06.00/share/sofa:/home/sofa/SOFA/v23.06.00/share/sofa/examples
[INFO]    [runSofa] GuiDataRepository paths = /home/sofa/SOFA/v23.06.00/share/sofa/gui/runSofa:/home/sofa/SOFA/v23.06.00/share/sofa/gui/qt
[INFO]    [GUIManager] Registered batch as a GUI.
[INFO]    [runSofa] Loading automatically plugin list in /home/sofa/SOFA/v23.06.00/lib/plugin_list.conf
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/lib/libSofa.Component.ODESolver.Forward.so
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/lib/libSofa.Component.ODESolver.Backward.so
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/lib/libSofa.Component.ODESolver.so
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/lib/libSofa.Component.IO.Mesh.so
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/lib/libSofa.Component.IO.so
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/lib/libSofa.Component.Playback.so
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/lib/libSofa.Component.SceneUtility.so
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/lib/libSofa.Component.Topology.Container.Constant.so
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/lib/libSofa.Component.Topology.Container.Dynamic.so
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/lib/libSofa.Component.Topology.Container.Grid.so
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/lib/libSofa.Component.Topology.Container.so
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/lib/libSofa.Component.Topology.Mapping.so
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/lib/libSofa.Component.Topology.Utility.so
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/lib/libSofa.Component.Topology.so
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/lib/libSofa.Component.Visual.so
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/lib/libSofa.Component.LinearSystem.so
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/lib/libSofa.Component.LinearSolver.Iterative.so
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/lib/libSofa.Component.LinearSolver.Direct.so
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/lib/libSofa.Component.LinearSolver.Preconditioner.so
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/lib/libSofa.Component.LinearSolver.so
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/lib/libSofa.Component.Mass.so
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/lib/libSofa.Component.Diffusion.so
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/lib/libSofa.Component.Mapping.Linear.so
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/lib/libSofa.Component.Mapping.NonLinear.so
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/lib/libSofa.Component.Mapping.MappedMatrix.so
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/lib/libSofa.Component.Mapping.so
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/lib/libSofa.Component.StateContainer.so
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/lib/libSofa.Component.SolidMechanics.Spring.so
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/lib/libSofa.Component.SolidMechanics.FEM.Elastic.so
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/lib/libSofa.Component.SolidMechanics.FEM.HyperElastic.so
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/lib/libSofa.Component.SolidMechanics.FEM.NonUniform.so
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/lib/libSofa.Component.SolidMechanics.FEM.so
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/lib/libSofa.Component.SolidMechanics.TensorMass.so
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/lib/libSofa.Component.SolidMechanics.so
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/lib/libSofa.Component.Constraint.Lagrangian.Model.so
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/lib/libSofa.Component.Constraint.Lagrangian.Correction.so
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/lib/libSofa.Component.Constraint.Lagrangian.Solver.so
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/lib/libSofa.Component.Constraint.Lagrangian.so
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/lib/libSofa.Component.Constraint.Projective.so
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/lib/libSofa.Component.Constraint.so
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/lib/libSofa.Component.AnimationLoop.so
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/lib/libSofa.Component.MechanicalLoad.so
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/lib/libSofa.Component.Collision.Geometry.so
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/lib/libSofa.Component.Collision.Detection.Algorithm.so
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/lib/libSofa.Component.Collision.Detection.Intersection.so
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/lib/libSofa.Component.Collision.Detection.so
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/lib/libSofa.Component.Collision.Response.Mapper.so
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/lib/libSofa.Component.Collision.Response.Contact.so
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/lib/libSofa.Component.Collision.Response.so
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/lib/libSofa.Component.Collision.so
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/lib/libSofa.Component.Setting.so
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/lib/libSofa.Component.Controller.so
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/lib/libSofa.Component.Engine.Analyze.so
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/lib/libSofa.Component.Engine.Generate.so
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/lib/libSofa.Component.Engine.Select.so
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/lib/libSofa.Component.Engine.Transform.so
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/lib/libSofa.Component.Engine.so
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/lib/libSofa.Component.Haptics.so
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/lib/libSofa.Component.so
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/lib/libSofa.GL.Component.Rendering2D.so
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/lib/libSofa.GL.Component.Rendering3D.so
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/lib/libSofa.GL.Component.Engine.so
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/lib/libSofa.GL.Component.Shader.so
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/lib/libSofa.GL.Component.so
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/lib/libSofa.GUI.Component.so
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/lib/libSofa.GUI.Batch.so
[INFO]    [GUIManager] Registered qglviewer as a GUI.
[INFO]    [GUIManager] Registered qt as a GUI.
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/lib/libSofa.GUI.Qt.so
[DEPRECATED] [SofaValidation] SofaValidation is deprecated. It will be removed at v23.06. For CompareState and CompareTopology, use Sofa.Component.Playback instead.
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/plugins/SofaValidation/lib/libSofaValidation.so
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/plugins/SceneCreator/lib/libSceneCreator.so
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/plugins/CImgPlugin/lib/libCImgPlugin.so
[ERROR]   [PluginManager] Plugin not found: "MeshSTEPLoader"
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/plugins/SofaMatrix/lib/libSofaMatrix.so
[INFO]    [PluginManager] Loaded plugin: /home/sofa/SOFA/v23.06.00/plugins/BeamAdapter/lib/libBeamAdapter.so
[INFO]    [PluginManager] 72 plugins have been loaded from /home/sofa/SOFA/v23.06.00/lib/plugin_list.conf
[INFO]    [SofaPluginManager] Loading automatically plugin list in /home/sofa/SOFA/v23.06.00/config/loadedPlugins.ini
[INFO]    [PluginManager] 0 plugins have been loaded from /home/sofa/SOFA/v23.06.00/config/loadedPlugins.ini

[INFO]    [SceneCheckerVisitor] Validating node "root" with checks: [SceneCheckCollisionResponse, SceneCheckDeprecatedComponents, SceneCheckDuplicatedName, SceneCheckMissingRequiredPlugin, SceneCheckUsingAlias]
[INFO]    [SceneCheckerVisitor] Finished validating node "root".


[INFO]    [SceneCheckerVisitor] Validating node "root" with checks: [SceneCheckCollisionResponse, SceneCheckDeprecatedComponents, SceneCheckDuplicatedName, SceneCheckMissingRequiredPlugin, SceneCheckUsingAlias]
[INFO]    [SceneCheckerVisitor] Finished validating node "root".
